### ISSUE TITLE
Detect sandbagging 

### ISSUE THREAD

**Issue by dlehdgus99** [online-go/online-go.com#2332]
**Is your feature request related to a problem? Please describe.**
I've noticed that some players resign even though they are clearly winning so that they can lower their rank for sandbagging.

**Describe the solution you'd like**
We can detect this by checking the win percentage provided by the AI. If a player repeteadly resigns games with over 90% winning rate we can automatically report this player for sandbagging so that moderators can take a look. I would love to be able to work on this if the issue is approved.

**Comment by GreenAsJade** [comment]
Yeah - just confirming this is an idea that's in "The list" of "things we really ought to do".

**Comment by dlehdgus99** [comment]
Thanks for the response. Please let me know if this idea is indeed in "The list". I would love to be able to start implementing the new feature.

**Comment by BenWestgate** [comment]
https://online-go.com/player/1629353/

An issue from using a "combined" rating for match making allows a sandbagger to have his cake (an accurate rating at least vs weaker players) and eat it too:

<img width="1216" height="616" alt="Image" src="https://github.com/user-attachments/assets/66646971-0f5d-457c-a719-cd864b07a5f0" />

<img width="1216" height="616" alt="Image" src="https://github.com/user-attachments/assets/c68a9a6f-5511-49da-95c4-33a566ec70f3" />

He simply resigns or quickly loses a blitz 9x9 game enough times to keep his overall rank at 10k so he can continue having a 91% win rate in even games against ddk's even though his performance is about 1 kyu and should be playing people his own strength or handicap games that are challenging for both.

**Comment by GreenAsJade** [comment]
Please report the person for sandbagging.

It's remarkably hard to automatically detect this (so many ifs and buts) - better to let moderators handle it.

**Comment by BenWestgate** [comment]
A few rules that would work for automatic detection:

1. Nobody has ratings in the different board sizes confidently 14-20+ ranks apart.
2. Nobody should have the majority of their losses be in positions the AI is not certain is lost.
3. Nobody should have the majority of their losses in the first 15, 30 or 60 turns (9x9, 13x13 and 19x19 respectively).
4. Nobody should have the majority of their resignations be in positions the AI is certain is a win.
5. Nobody better than 25 kyu overall should lose 25-50 games in a row in any board size.
6. Nobody worse than 1st Dan overall should win 25-50 games in a row in any board size.

If more than a couple of these are true, the player should be flagged.

I've reported the person and they're still doing it.

<img width="801" height="1350" alt="Image" src="https://github.com/user-attachments/assets/9c90a0e1-a7c2-48b0-aade-9891b5fa8bd8" />

<img width="782" height="784" alt="Image" src="https://github.com/user-attachments/assets/a3e16903-820f-4ae9-b232-d2d05226a249" />
